go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+	assert.Contains(t, reflect.ValueOf(destinationMasterTabletVars["TabletStateName"]).String(), "NOT_SERVING")
+
+	// check we can't migrate the master just yet
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("MigrateServedFrom", "destination_keyspace/0", "master")
+	assert.NotNil(t, err)
+
+	// migrate rdonly only in test_ny cell, make sure nothing is migrated
+	// in test_nj
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("MigrateServedFrom", "--cells=test_ny", "destination_keyspace/0", "rdonly")
+	assert.Nil(t, err)
+
+	// check SrvKeyspace
+	checkSrvKeyspaceServedFrom(t, cellj, destinationKeyspace, ksServedFrom, *clusterInstance)
+	checkBlacklistedTables(t, sourceMasterTablet, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceReplicaTablet, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceRdOnlyTablet1, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceRdOnlyTablet2, sourceKeyspace, nil)
+
+	// migrate test_nj only, using command line manual fix command,
+	// and restore it back.
+	keyspaceJSON, err := clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetKeyspace", "destination_keyspace")
+	assert.Nil(t, err)
+
+	var keyspace topodata.Keyspace
+	err = json2.Unmarshal([]byte(keyspaceJSON), &keyspace)
+	assert.Nil(t, err)
+	found := false
+	for _, servedFrom := range keyspace.GetServedFroms() {
+		if strings.ToLower(servedFrom.GetTabletType().String()) == "rdonly" {
+			found = true
+			assert.Contains(t, strings.Join(servedFrom.GetCells(), " "), "test_ca")
+			assert.Contains(t, strings.Join(servedFrom.GetCells(), " "), "test_nj")
+		}
+	}
+	assert.Equal(t, true, found)
+
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("SetKeyspaceServedFrom", "-source=source_keyspace", "-remove", "-cells=test_nj,test_ca", "destination_keyspace", "rdonly")
+	assert.Nil(t, err)
+
+	// again validating keyspaceJSON
+	keyspaceJSON, err = clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetKeyspace", "destination_keyspace")
+	assert.Nil(t, err)
+
+	err = json2.Unmarshal([]byte(keyspaceJSON), &keyspace)
+	assert.Nil(t, err)
+	found = false
+	for _, servedFrom := range keyspace.GetServedFroms() {
+		if strings.ToLower(servedFrom.GetTabletType().String()) == "rdonly" {
+			found = true
+		}
+	}
+	assert.Equal(t, false, found)
+
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("SetKeyspaceServedFrom", "-source=source_keyspace", "destination_keyspace", "rdonly")
+	assert.Nil(t, err)
+
+	keyspaceJSON, err = clusterInstance.VtctlclientProcess.ExecuteCommandWithOutput("GetKeyspace", "destination_keyspace")
+	assert.Nil(t, err)
+
+	err = json2.Unmarshal([]byte(keyspaceJSON), &keyspace)
+	assert.Nil(t, err)
+	found = false
+	for _, servedFrom := range keyspace.GetServedFroms() {
+		if strings.ToLower(servedFrom.GetTabletType().String()) == "rdonly" {
+			found = true
+			assert.Equal(t, []string{}, servedFrom.GetCells())
+		}
+	}
+	assert.Equal(t, true, found)
```

Comment:
Duplicate code, may be easier to move in a separate method.

Reply:
Done. Refer [PR46](https://github.com/planetscale/vitess/pull/46)